### DISCUSSION TITLE
How to debug a "already borrowed: BorrowMutError" error?

### DISCUSSION THREAD

**Discussion by frnsys** [leptos-rs/leptos#2673] [Q&A]
I'm struggling to figure out how I should debug a "already borrowed: BorrowMutError":

```
panicked at /home/francis/.cargo/registry/src/index.crates.io-6f17d22bba15001f/leptos_reactive-0.6.12/src/signal.rs:2082:39:
already borrowed: BorrowMutError
```

The traceback in the browser console isn't very helpful as it just points into the WASM and not anywhere in the original Rust. I'm at a loss about where I should look to figure out what's causing this conflict.

Thanks

**Comment by gbj** [comment]
Hard to know with so little context. Likely possibilities would be accidentally nesting a signal update inside another signal update, etc. My typical debugging for an "I have no idea what's happening" issue would just be to start commenting out sections of what I'm doing until it works, then see where it breaks. Happy to help debug with any additional information.

**Comment by frnsys** [comment]
Thanks, that's how I ended up figuring out the problem. I was mostly interested to know if there was any tooling to streamline the process, but for now this approach works. Thanks!
